linkis-computation-governance/linkis-computation-governance-common/src/main/scala/org/apache/linkis/governance/common/paser/CodeParser.scala

```diff
@@ -300,12 +302,18 @@ object CodeType extends Enumeration {
   type CodeType = Value
   val Python, SQL, Scala, Shell, Other, Remain, JSON = Value
 
-  def getType(codeType: String): CodeType = codeType.toLowerCase() match {
-    case "python" | "pyspark" | "py" => Python
-    case "sql" | "hql" | "psql" => SQL
-    case "scala" => Scala
-    case "shell" | "sh" => Shell
-    case _ => Other
+  def getType(codeType: String): CodeType = {
+    val runTypeAndCodeTypeRelationMap: Map[String, String] = CodeAndRunTypeUtils.getRunTypeAndCodeTypeRelationMap
+    if (runTypeAndCodeTypeRelationMap.isEmpty) return Other
```

Comment:
Is this better?   
`if (runTypeAndCodeTypeRelationMap.isEmpty || runTypeAndCodeTypeRelationMap.contains(codeType) ) return Other`